code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/unit/DeploymentUnitService.java

```diff
@@ -115,19 +116,29 @@ public Service getService() {
     }
 
     @SuppressWarnings("unchecked")
-    public List<String> getSidekickRefs(String launchConfigName) {
-        List<String> sidekicksLaunchConfigIds = new ArrayList<>();
+    public List<String> getSidekickRefs(Service service, String launchConfigName) {
+        List<String> configNames = new ArrayList<>();
         for (DeploymentUnit.SidekickType sidekickType : DeploymentUnit.SidekickType.supportedTypes) {
             Object sidekicksLaunchConfigObj = ServiceDiscoveryUtil.getLaunchConfigObject(service, launchConfigName,
                     sidekickType.launchConfigType);
             if (sidekicksLaunchConfigObj != null) {
                 if (sidekickType.isList) {
-                    sidekicksLaunchConfigIds.addAll((List<String>) sidekicksLaunchConfigObj);
+                    configNames.addAll((List<String>) sidekicksLaunchConfigObj);
                 } else {
-                    sidekicksLaunchConfigIds.add(sidekicksLaunchConfigObj.toString());
+                    configNames.add(sidekicksLaunchConfigObj.toString());
                 }
             }
         }
-        return sidekicksLaunchConfigIds;
+
+        List<String> toReturn = new ArrayList<>();
+        for (String name : configNames) {
+            if (name.equalsIgnoreCase(service.getName())) {
```

Comment:
With sidekicks, when do networkFrom/volumesFrom from sidekick to master, we accept service name. But internally service name gets translated to ServiceDiscoveryConstants.PRIMARY_LAUNCH_CONFIG_NAME to support service rename situation. There was one place where we didn't make this conversion.